pennylane_lightning/lightning_gpu/_state_vector.py

```diff
@@ -37,11 +70,308 @@ class LightningGPUStateVector(LightningBaseStateVector):
         device_name(string): state vector device name. Options: ["lightning.gpu"]
     """
 
-    def __init__(self, num_wires, dtype=np.complex128, device_name="lightning.gpu"):
-
-        if device_name != "lightning.gpu":
-            raise DeviceError(f'The device name "{device_name}" is not a valid option.')
+    def __init__(
+        self,
```

Comment:
Do we want to add docstring for the new args?

Reply:
done